src/material-experimental/mdc-form-field/form-field.ts

```diff
@@ -243,6 +243,11 @@ export class MatFormField implements AfterViewInit, OnDestroy, AfterContentCheck
     // have the ability to control the floating label state (i.e. `shouldLabelFloat`), and we
     // want to update the notch whenever the `_shouldLabelFloat()` value changes.
     getLabelWidth: () => 0,
+
+    // TODO: MDC now supports setting the required asterisk marker directly on
+    // the label component. This adapter method may be implemented and
+    // mat-mdc-form-field-required-marker removed.
```

Comment:
Is the plan that we clean this up as a follow-up? I don't think we'll use that given that we support custom form field controls and/or their associated ng control.

Reply:
I think it's optional to clean up.

I didn't do a deep dive into the component, but it looks like the `<span class="mat-mdc-form-field-required-marker">` accomplishes the same goal.

Instead of doing it in markup, you could do it in this method:

```ts
setLabelRequired(isRequired: boolean) {
  if (!this.hideRequiredMarker && this._control.required && !this.control.disabled) {
    // toggles the `mdc-floating-label--required` class
    this._floatingLabel.required = isRequired;
  }
}
```

Definitely a lot of opinions and unknowns on my part. Feel free to update the comment and either implement something, or leave it as no-op.

Reply:
Makes sense. I don't think we'll be able to rely on that method since we want to update whenever the input is refreshed, so we rely on change detection. Doing that manually might be overly complicated